### PR TITLE
IA-1299 Enforce minimum disk size of 50

### PR DESCRIPTION
**Pull request by gpcarr**
When we decode both GCE and Dataproc config, we are now enforcing a minimum disk size of 50GB. These changes reflect that and add two unit tests for validation.
---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala [hunk 1]
```diff
@@ -134,7 +134,7 @@ object JsonCodec {
   implicit val urlDecoder: Decoder[URL] =
     Decoder.decodeString.emap(s => Either.catchNonFatal(new URL(s)).leftMap(_.getMessage))
   implicit val diskSizeDecoder: Decoder[DiskSize] =
-    Decoder.decodeInt.emap(d => if (d < 0) Left("Negative number is not allowed") else Right(DiskSize(d)))
+    Decoder.decodeInt.emap(d => if (d < 50) Left("Minimum disk size is 50GB.") else Right(DiskSize(d)))
```
Comment (gpcarr) [inline core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala]: @Qi77Qi would we want to keep the "Negative number is not allowed" error and add a new one for minimum disk size or replace it like I've done in this PR? looking at fixing unit tests now and want to make sure I don't get rid of functionality that we may still want

Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala]: I think just erroring min disk is 50GB is clear enough...thanks for the quick fix 👍 ...maybe reword this to `Minimum required disk size is 50GB.`?

Just to double check, this is true for GCE VMs runtimes as well? cc @rtitle  for confirmation

**Review by Qi77Qi — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala (hunk 1 above)
Comment (gpcarr) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodec.scala]: Sounds good, will update! thank you :)

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodecSpec.scala [hunk 2]
```diff
@@ -67,7 +67,24 @@ class LeoRoutesJsonCodecSpec extends FlatSpec with Matchers {
       json <- io.circe.parser.parse(inputString)
       r <- json.as[RuntimeConfigRequest.DataprocConfig]
     } yield r
-    decodeResult.leftMap(_.getMessage) shouldBe Left("Negative number is not allowed: DownField(masterDiskSize)")
+    decodeResult.leftMap(_.getMessage) shouldBe Left("Minimum required disk size is 50GB: DownField(masterDiskSize)")
+  }
+
+  it should "fail with minimumDiskSizeDecodingFailure when masterDiskSize is less than 50" in {
```
Comment (Qi77Qi) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodecSpec.scala]: this is fine too...I would probably add the test in https://github.com/DataBiosphere/leonardo/blob/develop/core/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodecSpec.scala#L8 for just `diskSizeDecoder`

Comment (gpcarr) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodecSpec.scala]: I added `minimumDiskSizeDecodingFailure`  because we use the `negativeNumberDecodingFailure` in other places. and good point, I'll move my test now!

**Review by rtitle — APPROVED** [review]
looks good to me!

**Comment by rtitle** [comment]
btw we usually put the JIRA ticket number in the PR description -- we have a process to generate release notes based on commit logs.

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodecSpec.scala (hunk 2 above)
Comment (gpcarr) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodecSpec.scala]: @Qi77Qi I moved my unit test to `JsonCodecSpec` and have a unit test for a GCE config and a Dataproc config!

**Comment by gpcarr** [comment]
> btw we usually put the JIRA ticket number in the PR description -- we have a process to generate release notes based on commit logs.

Updated the PR description :)

File: core/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodecSpec.scala [hunk 3]
```diff
@@ -49,4 +53,39 @@ class JsonCodecSpec extends LeonardoTestSuite with Matchers with FlatSpecLike {
...
+      """
+        |{
+        |   "machineType": "n1-standard-8",
+        |   "diskSize": 35,
+        |   "cloudService": "GCE"
+        |}
+        |""".stripMargin
+
+    val res = decode[RuntimeConfig](inputString)
+    res.left shouldBe LeftProjection(Left(DecodingFailure("Minimum required disk size is 50GB", List(DownField("diskSize")))))
```
Comment (Qi77Qi) [inline core/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/JsonCodecSpec.scala:68]: btw, you should be able to do the check like this
```
res shouldBe Left(DecodingFailure("Minimum required disk size is 50GB", List(DownField("diskSize"))))
```
